spec/std/weak_ref_spec.cr

```diff
@@ -0,0 +1,70 @@
+require "spec"
+
+class State
```

Comment:
file-private classes are a general solution, I don't understand why we should go with a less general solution. I also want to use it [here](https://github.com/crystal-lang/crystal/blob/master/src/kernel.cr#L93), and this is not a spec. Sometimes there are some classes that only makes sense in a file, just like some functions that only make sense in a single file.